core/src/main/scala/kafka/server/KafkaConfig.scala

```diff
@@ -417,6 +423,10 @@ object KafkaConfig {
   val NumRecoveryThreadsPerDataDirDoc = "The number of threads per data directory to be used for log recovery at startup and flushing at shutdown"
   val AutoCreateTopicsEnableDoc = "Enable auto creation of topic on the server"
   val MinInSyncReplicasDoc = "define the minimum number of replicas in ISR needed to satisfy a produce request with acks=all (or -1)"
+  val MessageFormatVersionDoc = "Specify the message format version the broker will use to append messages to the logs."
```

Comment:
@gwenshap had some comments on the KIP about this. I feel fine either way. 

The good thing about piggybacking ApiVersion is it is easy to validate the config. And from user's perspective it may be easier to understand. e.g. During upgrade they can simply put their previous Kafka version there and after upgrade they just need to change it to the upgraded version. So users don't need to remember the magic values to put.